includes/class-newspack-app-shell.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Newspack App Shell
+ *
+ * @package Newspack_App_Shell
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main Newspack App Shell Class.
+ */
+final class Newspack_App_Shell {
+	const NEWSPACK_APP_SHELL_CPT = 'app_shell_cpt';
+
+	/**
+	 * The single instance of the class.
+	 *
+	 * @var Newspack_App_Shell
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Main Newspack App Shell Instance.
+	 * Ensures only one instance of Newspack App Shell is loaded or can be loaded.
+	 *
+	 * @return Newspack App Shell - Main instance.
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Register WP hooks.
+	 */
+	public function __construct() {
+		add_action( 'wp_enqueue_scripts', array( __CLASS__, 'enqueue_script' ) );
+		add_action( 'admin_menu', array( __CLASS__, 'add_menu_item' ) );
+		add_action( 'init', array( __CLASS__, 'register_cpt' ) );
+		add_action( 'wp_footer', array( __CLASS__, 'insert_persistent_content' ) );
+	}
+
+	/**
+	 * Register custom post type to store persistent element.
+	 */
+	public static function register_cpt() {
+		$cpt_args = array(
+			'labels'       => array(),
```

Comment:
```suggestion
			'labels'       => [
                     'singular_name'  => _x( 'Persistent element', 'Post type singular name', 'newspack-app-shell' ),
			         // etc.
			],
```
Doesn't necessarily need to be part of this PR, but you should add some labels at some point so it doesn't say "Edit post", "Post updated", etc. while working with the CPT and instead says "Edit persistent element", "Persistent element updated", etc. [Here are all the labels you can define](https://developer.wordpress.org/reference/functions/register_post_type/#comment-351).

Also, just as a fun fact we can use the array short notation (`[]`) now that the minimum PHP version for WP is >=5.6.

Reply:
I can't see a possibility to change this `Post updated.` message via `'labels'`. In theory it should be possible via [`post_updated_messages` hook](https://developer.wordpress.org/reference/hooks/post_updated_messages/), but I could not make it work. I've opened an issue so it can be tackled in the future: #7